### ISSUE TITLE
Fix S2259 FP: Property conditional is null check

### ISSUE THREAD

**Issue by snakefoot** [SonarSource/sonar-dotnet#7136]
### Description

S2259 - is null on at least one execution path

### Repro steps

```c#
        private ITypedLayout _typedDefaultValue;

        private object CreateTypedDefaultValue()

        {

            if (_typedDefaultValue?.IsFixed == true)

                return _typedDefaultValue
.FixedObjectValue;
        }
### Expected behavior

Not reported as bug

### Actual behavior

Reported as bug

### Known workarounds

Rewrite code

### Related information

https://sonarcloud.io/project/issues?resolved=false&types=BUG&branch=dev&id=nlog2&open=AYfDq4u9bnW2vP7_L8Kx

**Comment by snakefoot** [comment]
Maybe duplicate of #7104 ?

**Comment by pavel-mikula-sonarsource** [comment]
Hi @snakefoot,

Thank you for reporting it. Yes, it's a duplicate of #7104  that you can track. I'll add your test case there.
